modules/videoio/src/cap_mfx_common.cpp

```diff
@@ -14,10 +14,12 @@
 using namespace std;
 using namespace cv;
 
+static size_t IMPL = utils::getConfigurationParameterSizeT("OPENCV_VIDEOIO_MFX_IMPL", MFX_IMPL_AUTO_ANY);
```

Comment:
Lets avoid exploding of global static initialization tables.
Use lazy on-demand initialization instead. See `static const String getBuildExtraOptions()` from ocl.cpp

Reply:
Should I repeat scheme with `initialized` flag or use simple static initialization like shown below?
```.cpp
size_t getValue()
{
    static const size_t value = initValue();
    return value;
}
```

Reply:
Something like this is enough (sample code prints "overloaded" message):
```
size_t getValue()
{
    static const size_t value = utils::getConfigurationParameterSizeT("OPENCV_VIDEOIO_MFX_IMPL", MFX_IMPL_AUTO_ANY);
    return value;
}
```